unison-core/src/Unison/Names.hs

```diff
@@ -102,6 +107,43 @@ instance Show (Names) where
       ++ foldMap (\(n, r) -> "  " ++ show n ++ " -> " ++ show r ++ "\n") (R.toList types)
       ++ "\n"
 
+-- | Abbreviate all names to their minimal unique suffix.
+--
+-- E.g.
+-- ["base.List.map", "base.Bag.map"] -> ["List.map", "Bag.map"]
+-- ["base.List.filter", "base.List.map"] -> ["filter", "map"]
+minimalUniqueSuffix :: Names -> Names
```

Comment:
I initially thought I would need this, but it's easier to handle within the PPE layer so this is currently unused.

I'd prefer to just leave it here since it may be useful in the future, but happy to remove it if it seems clutter-full